src/accessors/merge_reveal.rs

```diff
@@ -188,12 +188,19 @@ impl MergeReveal for TransitionBundle {
     fn merge_reveal(mut self, other: Self) -> Result<Self, MergeRevealError> {
         debug_assert_eq!(self.commitment_id(), other.commitment_id());
 
-        if self
-            .known_transitions
-            .extend(other.known_transitions)
-            .is_err() ||
-            self.input_map.len() > self.known_transitions.len()
-        {
+        for (opid, transition) in self.known_transitions.keyed_values_mut() {
+            if let Some((_, other_transition)) =
+                other.known_transitions.iter().find(|(o, _)| o == &opid)
```

Comment:
We can use `other.known_transitions.remove(opid)` and by doing that avoid clone below

Reply:
`other` is not mutable so I don't think calling `remove` is possible (but it was a good point, thanks)

Reply:
Other is here by value, meaning you can just add `mut` in front of it in the method signature

Reply:
You're right. I've updated the PR, thanks again

Reply:
Actually, I've tried this change on rgb-lib and it seems that now another test (where a wallet sends back what it has received) fails. So I'm going to revert this change.

Reply:
For more context, using `remove` instead of `find` makes the receiver fail the command `consignment.validate` with the error `consignment is invalid and can't be imported.` when the command is performed a second time (I guess because `remove` is removing data from `self.anchors`).

Reply:
I'm not sure what you intend. `other` comes from the `consume_bundle` method in the`hoard` module:
```rust
pub fn consume_bundle(&mut self, bundle: TransitionBundle) -> Result<(), ConsumeError> {
    let bundle_id = bundle.bundle_id();
    match self.bundles.get_mut(&bundle_id) {
        Some(b) => *b = b.clone().merge_reveal(bundle)?,
        None => {
            self.bundles.insert(bundle_id, bundle)?;
        }
    }
    Ok(())
}
```
so by changing the signature of `merge_reveal` like so:
```rust
fn merge_reveal(self, mut other: Self) -> Result<Self, MergeRevealError> {
```
and calling `remove` we are changing `self.bundles`,  causing so the previously mentioned bug. Not sure why we should remove and then re-insert but maybe I'm missing something

Reply:
We remove from `other`, not `self`: `other.known_transitions.find` -> `other.known_transitions.remove`

Reply:
Ok, I got it, sorry. `transitions` come from `self` and not `other`.

Reply:
But you can still do `other.transition.assignments.merge_reveal(self...)` instead of `self.transition.assignments.merge_reveal(other...)` and at least save one clone.

Sorry, these clones are really very performance-affecting and it is nice to avoid them wherever is possible. Especially here, where the operation is `O(N)` with `N` is the number of transitions in a stash!

Self = stash
other = consignment

Reply:
Not sure I've done what you're asking, but I tried doing:
```rust
transition.assignments = other_transition
    .assignments
    .clone()
    .merge_reveal(transition.assignments)
    .unwrap();
```
but then I receive the error
```
move occurs because `transition.assignments` has type `Assignments<BlindSeal<TxPtr>>, which does not implement the `Copy` trait
```
And if I try removing the previous clone (the one on `other.transition.assignments`) I receive the error
```
move occurs because `other_transition.assignments` has type `Assignments<BlindSeal<TxPtr>>`, which does not implement the `Copy` trait
```

Reply:
I added commit showing what I have meant :)

Reply:
Great, thanks

Reply:
Can I merge this? Can you pls confirm that it works with your tests?

Reply:
I confirm it works. You can merge this, thank you